server/server_test.go

```diff
@@ -0,0 +1,16 @@
+package server
+
+import (
+	"github.com/stretchr/testify/assert"
```

Comment:
we don't use assert, just standard go tests

Reply:
That is really unfortunate. You're missing out!

But it does look like you are using assert:

```
$ git grep 'testify/assert'
fs/dtype_linux_test.go:  "github.com/stretchr/testify/assert"
log/context_test.go: "github.com/stretchr/testify/assert"
mount/lookup_test/lookup_linux_test.go:  "github.com/stretchr/testify/assert"
remotes/docker/scope_test.go:    "github.com/stretchr/testify/assert"
server/server_test.go:   "github.com/stretchr/testify/assert"
snapshot/storage/metastore_test.go:  "github.com/stretchr/testify/assert"
snapshot/testsuite/testsuite.go: "github.com/stretchr/testify/assert"
testutil/helpers_unix.go:    "github.com/stretchr/testify/assert"
```

Reply:
humm, i didn't know what

Reply:
The problem here is that it doesn't fail the unit test. In every other language, "assert" means "ensure this condition or die". Usually, we want the `require` package.

With `(testing.T).Helper()`, this is almost useless now. We can probably get rid of the dependency!

Reply:
It "fails" the unit test (as in, the test will be reported as failing), but it doesn't exit immediately. The testing framework for most languages don't have support for that behaviour. I'd be happy to change it to `require`, but it doesn't make a difference if there is nothing else after it.  I will always use `require` for conditions that need to exit, and `assert` for conditions that don't. 

Using the "fail but don't exit" assert can be really nice when there are multiple return values to check, because you get more details about the failure.

`T.Helper()` makes it a lot easier to write test helpers, but it doesn't handle printing a helpful and correct error message on failure.

A smaller library which provides similar or better error messages using `T.Helper()` would be great.